### PR TITLE
[Backport 1.x] Adds code coverage reporting (#1478)

### LINKED ISSUE #1339: onboard to Codecov for code coverage
**Is your feature request related to a problem? Please describe.**

There is an project initiative for code coverage. See examples
https://github.com/opensearch-project/opensearch-build/blob/main/.codecov.yml
https://github.com/opensearch-project/security-dashboards-plugin/blob/main/codecov.yml

**Describe the solution you'd like**

onboard to Codecov

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by manasvinibs**
* Renames build and test workflow to eliminate ambiguity.
* Adds coverage generation for jest and mocha tests.
* Expands coverage collection for jest tests.
* Utilizes `lcov` reporter as well as text summary reporter.

Resolves #1339
Prerequisite for #1035

Signed-off-by: Tommy Markley <markleyt@amazon.com>
(cherry picked from commit 1caf90752d8e34ceece81d8a69f6ec79aabfa9c1)

### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,13 +42,15 @@
     "opensearch": "node scripts/opensearch",
     "test": "grunt test",
     "test:jest": "node scripts/jest",
+    "test:jest:coverage": "node scripts/jest --coverage",
+    "test:jest:ci": "node scripts/jest --ci --colors --runInBand",
+    "test:jest:ci:coverage": "node scripts/jest --ci --colors --runInBand --coverage",
     "test:jest_integration": "node scripts/jest_integration",
     "test:mocha": "node scripts/mocha",
-    "test:mocha:coverage": "grunt test:mochaCoverage",
+    "test:mocha:coverage": "yarn nyc --reporter=text-summary --reporter=lcov --report-dir=./target/opensearch-dashboards-coverage/mocha node scripts/mocha",
```
Comment (AMoo-Miki) [inline package.json:50]: Do we know that these work in 1.x?

**Review by joshuarrrr — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (manasvinibs) [reply package.json:50]: When I tried running the above cmd, I got all the tests passing with below summary

```
411 passing (7s)


=============================== Coverage summary ===============================
Statements   : 45.54% ( 1956/4295 )
Branches     : 20.79% ( 431/2073 )
Functions    : 25.82% ( 228/883 )
Lines        : 46.59% ( 1908/4095 )
================================================================================
Done in 16.20s.```

**Comment by joshuarrrr** [comment]
looks like the bootstrap step failed on downloading geckodriver

**Comment by manasvinibs** [comment]
> looks like the bootstrap step failed on downloading geckodriver

I think it's flaky. In the last run, it was passing bootstrap https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/3962638107/jobs/6789549417#step:12:1232

**Comment by joshuarrrr** [comment]
> > looks like the bootstrap step failed on downloading geckodriver
> 
> I think it's flaky. In the last run, it was passing bootstrap https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/3962638107/jobs/6789549417#step:12:1232

yeah, that's usually the case when it errors on fetching some resource.

**Pull request closed without merging**